Code/SignHandler.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+
+namespace ArchipelagoRandomizer
+{
+	/// <summary>
+	/// Handles granting items when signs are read
+	/// </summary>
+	public class SignHandler : MonoBehaviour
+	{
+		private readonly List<SignData> allSignData = new();
+		private List<ItemRandomizer.LocationData.Location> allSignLocations;
+
+		public static SignHandler Instance { get; private set; }
+
+		/* NOT IN USE CURRENTLY, INHERIT ITEMRANDOCOMPONENT WHEN READY TO USE
```

Comment:
Is Item Rando Component not yet ready to use?

Reply:
It's still a work in progress and not used currently. That'll come after reworking some of the component initialization to better support it